source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -543,6 +578,71 @@ void CacheFilter::finalizeEncodingCachedResponse() {
   filter_state_ = FilterState::ResponseServedFromCache;
 }
 
+LookupStatus CacheFilter::lookupStatus() const {
+  if (lookup_result_ == nullptr && lookup_ != nullptr) {
+    return LookupStatus::RequestIncomplete;
+  }
+
+  if (lookup_result_ != nullptr) {
+    switch (lookup_result_->cache_entry_status_) {
+    case CacheEntryStatus::Ok:
+      return LookupStatus::CacheHit;
+    case CacheEntryStatus::Unusable:
+      return LookupStatus::CacheMiss;
+    case CacheEntryStatus::RequiresValidation:
+      // The CacheFilter sent the response upstream for validation; check the
+      // filter state to see whether and how the upstream responded. The
+      // filter currently won't send the stale entry if it can't reach the
+      // upstream or if the upstream responds with a 5xx, so don't include
+      // special handling for those cases.
+      switch (filter_state_) {
+      case FilterState::ValidatingCachedResponse:
+        return LookupStatus::RequestIncomplete;
+      case FilterState::EncodeServingFromCache:
+      case FilterState::ResponseServedFromCache:
+        return LookupStatus::StaleHitWithSuccessfulValidation;
+      case FilterState::NotServingFromCache:
+        return LookupStatus::StaleHitWithFailedValidation;
+      default:
+        IS_ENVOY_BUG(absl::StrCat("Unexpected filter state in requestCacheStatus: cache lookup "
+                                  "response required validation, but filter state is ",
+                                  filter_state_));
+        return LookupStatus::Unknown;
+      }
+    case CacheEntryStatus::FoundNotModified:
+      // TODO(capoferro): Report this as a FoundNotModified when we handle
+      // those.
+      return LookupStatus::CacheHit;
+    case CacheEntryStatus::LookupError:
+      return LookupStatus::LookupError;
+    }
+    IS_ENVOY_BUG(absl::StrCat(
+        "Unhandled CacheEntryStatus encountered when retrieving request cache status: " +
+        std::to_string(static_cast<int>(filter_state_))));
+    return LookupStatus::Unknown;
+  }
+
+  // Either decodeHeaders decided not to do a cache lookup (because the
+  // request isn't cacheable), or decodeHeaders hasn't been called yet.
+  switch (filter_state_) {
+  case FilterState::Initial:
+    return LookupStatus::RequestIncomplete;
+  case FilterState::NotServingFromCache:
+    return LookupStatus::RequestNotCacheable;
+  default:
```

Comment:
consider enumerating the valid states here rather than using a default, so someone adding a new state is forced to consider this switch statement.